### ISSUE TITLE
setDraggable does not work without adding a mouse joint

### ISSUE THREAD

**Issue by catmanjan** [flixel-gdx/flixel-gdx-box2d#2]
Not sure if this is how Box2D works, but adding
`add(new B2FlxMouseJoint());`

Makes setDraggable actually work. Why not just always add a mouse joint in B2FlxState?

**Comment by WingEraser** [comment]
I didn't added that because of performance. Once you click on a box2d object it will go through some calculations. You can see it in `B2FlxMouseJoint::mouseDown()`. It needs to query all fixtures that may overlap the provided AABB. After that it needs to know whether the body is draggable or not.

My game uses box2d with a lot of bodies. The controls are just FlxButtons, but every time I press on the screen, it doesn't have to query all the fixtures.

**Comment by catmanjan** [comment]
How about automatically adding a mouse joint (if one doesn't already exist) as soon as setDraggable is set to true on any body?

**Comment by WingEraser** [comment]
Nope, ain't going to happen. I don't want any specific joint to be in B2FlxShape. It will also have some problem with multitouch support (if I ever managed to get that working properly). I need multiple mouse joints for that.
